galsim/base.py

```diff
@@ -338,6 +338,258 @@ def getGSParams(self):
         """
         return self.SBProfile.getGSParams()
 
+    def calculateHLR(self, size=None, scale=None, centroid=None, flux_frac=0.5):
+        """Returns the half-light radius of the object.
+
+        If the profile has a half_light_radius attribute, it will just return that, but in the
```

Comment:
Same comment applies to the moment radius routine (though in that case, we have to make sure there's no noise to prevent divergence of the unweighted moments).